volumes.go

```diff
@@ -12,34 +12,31 @@ func (*Volume) ResourceType() string {
 	return "Volume"
 }
 
-// Get fetches the given volume by ID
+// Get fetches the given volume
 func (vol *Volume) Get(ctx context.Context, client *Client) error {
-	if vol.ID == "" && vol.VirtualMachineID == "" {
-		return fmt.Errorf("a Volume may only be get using ID or VirtualMachineID")
-	}
-
 	vols, err := client.ListWithContext(ctx, vol)
 	if err != nil {
 		return err
 	}
 
-	queryErrorStr := ""
-	if vol.ID != "" && vol.VirtualMachineID != "" {
-		queryErrorStr = fmt.Sprintf("id: %s VirtualMachineID: %s", vol.ID, vol.VirtualMachineID)
-	} else if vol.ID != "" {
-		queryErrorStr = fmt.Sprintf("id: %s", vol.ID)
-	} else {
-		queryErrorStr = fmt.Sprintf("VirtualMachineID: %s", vol.VirtualMachineID)
-	}
-
 	count := len(vols)
-	if count == 0 {
-		return &ErrorResponse{
-			ErrorCode: ParamError,
-			ErrorText: fmt.Sprintf("missing Volume. Query: %s", queryErrorStr),
+	if count != 1 {
+		req, err := vol.ListRequest()
+		if err != nil {
+			return err
+		}
+		payload, err := client.Payload(req)
+		if err != nil {
+			return err
+		}
+
+		if count == 0 {
+			return &ErrorResponse{
+				ErrorCode: ParamError,
+				ErrorText: fmt.Sprintf("missing Volume. query: %s", payload),
+			}
```

Comment:
@pierre-emmanuelJ this is more or less what I had in mind for #155

Reply:
does it help you with #155?

Reply:
yes